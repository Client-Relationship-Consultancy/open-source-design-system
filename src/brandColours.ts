export const colourPalette = {
  examplePalette: {
    primary: {
      main: {
        hex: "#A42824",
        RGB: "164,40,36",
        on: "#FFFFFF",
      },
      light: {
        hex: "#DA342F",
        RGB: "218,52,47",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#661614",
        RGB: "102,22,20",
        on: "#FFFFFF",
      },
    },
    secondary: {
      main: {
        hex: "#0B466B",
        RGB: "11,70,107",
        on: "#FFFFFF",
      },
      light: {
        hex: "#599CAF",
        RGB: "89,156,175",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#05263B",
        RGB: "5,38,59",
        on: "#FFFFFF",
      },
    },
    complimentary: {
      main: {
        hex: "#FFEE52",
        RGB: "255,238,82",
        on: "#272727",
      },
      light: {
        hex: "#FFF59D",
        RGB: "255,245,157",
        on: "#272727",
      },
      dark: {
        hex: "#CEB900",
        RGB: "206,185,0",
        on: "#272727",
      },
    },
    action: {
      main: {
        hex: "#743C8F",
        RGB: "116,60,143",
        on: "#FFFFFF",
      },
      light: {
        hex: "#AD65CF",
        RGB: "173,101,207",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#4E2662",
        RGB: "79,38,98",
        on: "#FFFFFF",
      },
    },
    error: {
      main: {
        hex: "#FF7320",
        RGB: "255,115,32",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#C54901",
        RGB: "197,73,1",
        on: "#FFFFFF",
      },
    },
    background: {
      hex: "#F6F6F6",
      RGB: "246,246,246",
    },
    surface: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    white: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    black: {
      main: { hex: "#272727", RGB: "39,39,39", on: "#FFFFFF" },
      tint80: { hex: "#696969", RGB: "105,105,105", on: "#FFFFFF" },
      tint60: { hex: "#A0A0A0", RGB: "160,160,160", on: "#FFFFFF" },
      tint40: { hex: "#C7C7C7", RGB: "199,199,199", on: "#272727" },
      tint20: { hex: "#ECECEC", RGB: "236,236,236", on: "#272727" },
    },
  },
};

interface IColourCodes {
  hex: string;
  RGB: string;
  on?: string;
}
interface IColourVariations {
  main: IColourCodes;
  light?: IColourCodes;
  dark?: IColourCodes;
}

interface IBlackVariations {
  main: IColourCodes;
  tint80: IColourCodes;
  tint60: IColourCodes;
  tint40: IColourCodes;
  tint20: IColourCodes;
}

export interface IColourPalette {
  primary: IColourVariations;
  secondary: IColourVariations;
  complimentary: IColourVariations;
  action: IColourVariations;
  error: IColourVariations;
  background: IColourCodes;
  surface: IColourCodes;
  white: IColourCodes;
  black: IBlackVariations;
}
