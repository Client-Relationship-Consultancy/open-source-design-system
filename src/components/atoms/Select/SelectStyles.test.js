import { colourPalette } from "brandColours"
import style from "./SelectStyles"

const { secondary, white, black } = colourPalette.examplePalette

describe("Test Select Renderer Styles", () => {
  it("Test dropdown indicator", () => {
    const dropdownIndicator = style(colourPalette.examplePalette).dropdownIndicator({
      test: "test",
    })
    expect(dropdownIndicator).toMatchSnapshot()
  })
  it("Test clear indicator", () => {
    const clearIndicator = style(colourPalette.examplePalette).clearIndicator({ test: "test" })
    expect(clearIndicator).toMatchSnapshot()
  })
  it("Test control is focused = true", () => {
    const control = style(colourPalette.examplePalette).control(
      { test: "test" },
      { isFocused: true },
    )
    expect(control).toMatchSnapshot()
  })
  it("Test control is focused = false", () => {
    const control = style(colourPalette.examplePalette).control(
      { test: "test" },
      { isFocused: false },
    )
    expect(control).toMatchSnapshot()
  })
  it("Test option method: isSelected = true, isFocused = false", () => {
    const optionReturn = style(colourPalette.examplePalette).option(
      { test: "test" },
      { isSelected: true, isFocused: false },
    )
    expect(optionReturn.test).toMatch("test")
    expect(optionReturn.backgroundColor).toMatch(secondary.dark.hex)
    expect(optionReturn.color).toMatch(white.hex)
  })
  it("Test option method: isSelected = false, isFocused = false", () => {
    const optionReturn = style(colourPalette.examplePalette).option(
      { test: "test" },
      { isSelected: false, isFocused: false },
    )
    expect(optionReturn.test).toMatch("test")
    expect(optionReturn.backgroundColor).toMatch(white.hex)
    expect(optionReturn.color).toMatch(black.main.hex)
  })
  it("Test option method: isSelected = true, isFocused = true", () => {
    const optionReturn = style(colourPalette.examplePalette).option(
      { test: "test" },
      { isSelected: true, isFocused: true },
    )
    expect(optionReturn.test).toMatch("test")
    expect(optionReturn.backgroundColor).toMatch(secondary.dark.hex)
    expect(optionReturn.color).toMatch(white.hex)
  })
  it("Test option method: isSelected = false, isFocused = true", () => {
    const optionReturn = style(colourPalette.examplePalette).option(
      { test: "test" },
      { isSelected: false, isFocused: true },
    )
    expect(optionReturn.test).toMatch("test")
    expect(optionReturn.backgroundColor).toMatch(secondary.main.hex)
    expect(optionReturn.color).toMatch(white.hex)
  })
  it("Test multiValue method", () => {
    const multiValueReturn = style(colourPalette.examplePalette).multiValue({ test: "test" })
    expect(multiValueReturn.test).toMatch("test")
    expect(multiValueReturn.backgroundColor).toMatch(secondary.main.hex)
  })
  it("Test multiValueLabel method", () => {
    const multiValueLabel = style(colourPalette.examplePalette).multiValueLabel({ test: "test" })
    expect(multiValueLabel.test).toMatch("test")
    expect(multiValueLabel.textAlign).toMatch("center")
  })
  it("Test multiValueRemove method", () => {
    const multiValueRemove = style(colourPalette.examplePalette).multiValueRemove({ test: "test" })
    expect(multiValueRemove.test).toMatch("test")
    expect(multiValueRemove.backgroundColor).toMatch(secondary.dark.hex)
  })
})
