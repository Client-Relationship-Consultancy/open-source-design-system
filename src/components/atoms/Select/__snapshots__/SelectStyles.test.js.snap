// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select Renderer Styles Test clear indicator 1`] = `
Object {
  ":hover": Object {
    "color": "#CEB900",
    "cursor": "pointer",
  },
  "color": "#FFEE52",
  "test": "test",
}
`;

exports[`Test Select Renderer Styles Test control is focused = false 1`] = `
Object {
  ":hover": Object {
    "cursor": "text",
  },
  ":hover, :active, :focus": Object {
    "border": "1px solid #05263B",
    "boxShadow": "0 0 0 1px #05263B",
  },
  "border": "1px solid #C7C7C7",
  "boxShadow": "none",
  "test": "test",
}
`;

exports[`Test Select Renderer Styles Test control is focused = true 1`] = `
Object {
  ":hover": Object {
    "cursor": "text",
  },
  ":hover, :active, :focus": Object {
    "border": "1px solid #05263B",
    "boxShadow": "0 0 0 1px #05263B",
  },
  "border": "1px solid #05263B",
  "boxShadow": "0 0 0 1px #05263B",
  "test": "test",
}
`;

exports[`Test Select Renderer Styles Test dropdown indicator 1`] = `
Object {
  ":hover": Object {
    "color": "#05263B",
    "cursor": "pointer",
  },
  "color": "#0B466B",
  "test": "test",
}
`;
